R/bs-theme.R

```diff
@@ -294,6 +294,9 @@ bootstrap_bundle <- function(version) {
           "offcanvas", "placeholders", "helpers", "utilities/api"
         ))
       ),
+      shiny = sass_layer(
+        mixins = sass_file(path_inst("shiny-scss", "bs5", "_variables.scss"))
+      ),
```

Comment:
Is this here (and not in the preset) because we think these styling decisions should generally apply to Bootstrap 5? In that case, it seems that this (as well as the rules below) should live in Shiny (but only applied when `$bootstrap-version` is a suitable value). 

Doing this should help prevent situations like https://github.com/rstudio/shiny/issues/3905, which still is a potential problem for other Shiny Sass like selectize.js. I was thinking we should provide something like a `shiny:::compileDependencies(theme)` that `PyShiny` could use to compile all of Shiny's Sass->CSS with whatever theme it wants.

Reply:
Yeah, in general I realized that these defaults improved all Bootstrap/Bootswatch 5 themes.

I'm not sure I agree that we should move them into Shiny, however. I don't think it would help with the linked issue (we misdiagnosed the problem when we thought it was about missing variables). 

My primary concern is that Bootstrap 5 comes from bslib and the more we push Bootstrap-version-specific rules and variables out of bslib, the further they are from the place that drives the change. Each incremental change we push out of bslib becomes one more place we have to remember to evaluate (or accidentally forget) for changes later, e.g. if there's another BS 5 patch release.

That said, maybe there's a wrinkle in how shiny notification styles are included from the Shiny side that I'm missing.

Reply:
This is now done in rstudio/shiny#3914. Once I removed it from bslib in this PR, however, we now enter a weird state where a newer bslib (i.e. with the changes in this PR) paired with an older version of Shiny won't have the right notification styles -- the colors will use the default shiny styles and not the newest variables.

I guess that's an intrinsic trade-off in moving the Bootstrap 5 defaults over to Shiny, but I wanted to call it out.